src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -432,6 +432,81 @@ begin
   simp only [set.mem_to_finset, mem_neighbor_set],
 end
 
+/-! ## Edge deletion -/
+
+/-- Given a subgraph `G'` and a set of vertex pairs, remove all of the corresponding edges
+from its edge set, if present.
+
+See also: `simple_graph.delete_edges`. -/
+def delete_edges (G' : G.subgraph) (s : set (sym2 V)) : G.subgraph :=
+{ verts := G'.verts,
+  adj := G'.adj \ sym2.to_rel s,
+  adj_sub := λ a b h', G'.adj_sub h'.1,
+  edge_vert := λ a b h', G'.edge_vert h'.1,
+  symm := λ a b, by simp [G'.adj_comm, sym2.eq_swap] }
+
+section delete_edges
+variables {G' : G.subgraph} (s : set (sym2 V))
+
+@[simp] lemma delete_edges_verts : (G'.delete_edges s).verts = G'.verts := rfl
+
+@[simp] lemma delete_edges_adj (v w : V) :
+  (G'.delete_edges s).adj v w ↔ G'.adj v w ∧ ¬ ⟦(v, w)⟧ ∈ s := iff.rfl
+
+@[simp] lemma delete_edges_delete_edges (s s' : set (sym2 V)) :
+  (G'.delete_edges s).delete_edges s' = G'.delete_edges (s ∪ s') :=
+by ext; simp [and_assoc, not_or_distrib]
+
+@[simp] lemma delete_edges_empty_eq : G'.delete_edges ∅ = G' :=
+by ext; simp
+
+@[simp] lemma delete_edges_spanning_coe_eq :
+  (G'.spanning_coe.delete_edges s) = (G'.delete_edges s).spanning_coe :=
+by { ext, simp }
+
+lemma delete_edges_coe_eq (s : set (sym2 G'.verts)) :
+  (G'.coe.delete_edges s) = (G'.delete_edges (sym2.map coe '' s)).coe :=
+begin
+  ext ⟨v, hv⟩ ⟨w, hw⟩,
+  simp only [simple_graph.delete_edges_adj, coe_adj, subtype.coe_mk, delete_edges_adj,
+    set.mem_image, not_exists, not_and, and.congr_right_iff],
+  intro h,
+  split,
+  { intros hs,
+    refine sym2.ind _,
+    rintro ⟨v', hv'⟩ ⟨w', hw'⟩,
+    simp only [sym2.map_pair_eq, subtype.coe_mk, quotient.eq],
+    contrapose!,
+    rintro (_ | _); simpa [sym2.eq_swap], },
+  { intros h' hs,
+    exact h' _ hs rfl, },
+end
+
+lemma delete_edges_le : G'.delete_edges s ≤ G' :=
+by split; simp { contextual := tt }
+
+lemma delete_edges_le_of_le {s s' : set (sym2 V)} (h : s ⊆ s') :
+  G'.delete_edges s' ≤ G'.delete_edges s :=
+begin
+  split;
+  simp only [delete_edges_verts, delete_edges_adj, true_and, and_imp] {contextual := tt},
+  exact λ v w hvw hs' hs, hs' (h hs),
+end
+
+lemma delete_edges_eq_delete_edges_inter_edge_set :
+  G'.delete_edges s = G'.delete_edges (s ∩ G'.edge_set) :=
```

Comment:
How about:
```suggestion
@[simp] lemma delete_edges_inter_edge_set_right_eq :
  G'.delete_edges (s ∩ G'.edge_set) = G'.delete_edges s :=
```
and also a "left" version?

Reply:
Good idea, thanks.